src/main/java/net/ccbluex/liquidbounce/injection/mixins/minecraft/entity/MixinLivingEntity.java

```diff
@@ -139,13 +144,17 @@ private Vec3d hookFixRotation(Vec3d instance, double x, double y, double z) {
         return instance.add(-MathHelper.sin(yaw) * 0.2F, 0.0, MathHelper.cos(yaw) * 0.2F);
     }
 
+
     @Inject(method = "pushAwayFrom", at = @At("HEAD"), cancellable = true)
-    private void hookNoPush(CallbackInfo callbackInfo) {
+    private void hookNoPush(Entity entity, CallbackInfo ci) {
         if (ModuleNoPush.INSTANCE.getEnabled()) {
-            callbackInfo.cancel();
+            // Checking if the entity is a playerEntity or a livingEntity depending if the playerOnly is toggled or not.
+            if ((entity instanceof PlayerEntity && ModuleNoPush.INSTANCE.getPlayerOnly()) ||
+                    (entity instanceof LivingEntity && !ModuleNoPush.INSTANCE.getPlayerOnly())) {
+                ci.cancel();
+            }
```

Comment:
This should be it's own PR

Reply:
Yes i already made a new pull request for it. 
The reason why this happened is because i didn't know how to make a new pull request sadly 😭 
But I've figured it out.

> This should be it's own PR